lib/matplotlib/backends/_backend_gtk.py

```diff
@@ -0,0 +1,183 @@
+        # never be stopped.
+        self._timer_stop()
+        self._timer = GLib.timeout_add(self._interval, self._on_timer)
+
+    def _timer_stop(self):
+        if self._timer is not None:
+            GLib.source_remove(self._timer)
+            self._timer = None
+
+    def _timer_set_interval(self):
+        # Only stop and restart it if the timer has already been started.
+        if self._timer is not None:
+            self._timer_stop()
+            self._timer_start()
+
+    def _on_timer(self):
+        super()._on_timer()
+
+        # Gtk timeout_add() requires that the callback returns True if it
+        # is to be called again.
+        if self.callbacks and not self._single:
+            return True
+        else:
+            self._timer = None
+            return False
+
+
+class _NavigationToolbar2GTK(NavigationToolbar2):
+    # Must be implemented in GTK3/GTK4 backends:
+    # * __init__
+    # * save_figure
+
+    def set_message(self, s):
+        escaped = GLib.markup_escape_text(s)
+        self.message.set_markup(f'<small>{escaped}</small>')
+
+    def draw_rubberband(self, event, x0, y0, x1, y1):
+        height = self.canvas.figure.bbox.height
+        y1 = height - y1
+        y0 = height - y0
+        rect = [int(val) for val in (x0, y0, x1 - x0, y1 - y0)]
+        self.canvas._draw_rubberband(rect)
+
+    def remove_rubberband(self):
+        self.canvas._draw_rubberband(None)
+
+    def _update_buttons_checked(self):
+        for name, active in [("Pan", "PAN"), ("Zoom", "ZOOM")]:
+            button = self._gtk_ids.get(name)
+            should_be_active = self.mode.name == active
+            if button and button.get_active() != should_be_active:
+                button.set_active(should_be_active)
+
+    def pan(self, *args):
+        super().pan(*args)
+        self._update_buttons_checked()
+
+    def zoom(self, *args):
+        super().zoom(*args)
+        self._update_buttons_checked()
+
+    def set_history_buttons(self):
+        can_backward = self._nav_stack._pos > 0
+        can_forward = self._nav_stack._pos < len(self._nav_stack._elements) - 1
+        if 'Back' in self._gtk_ids:
+            self._gtk_ids['Back'].set_sensitive(can_backward)
+        if 'Forward' in self._gtk_ids:
+            self._gtk_ids['Forward'].set_sensitive(can_forward)
+
+
+class RubberbandGTK(backend_tools.RubberbandBase):
+    def draw_rubberband(self, x0, y0, x1, y1):
+        _NavigationToolbar2GTK.draw_rubberband(
+            self._make_classic_style_pseudo_toolbar(), None, x0, y0, x1, y1)
+
+    def remove_rubberband(self):
+        _NavigationToolbar2GTK.remove_rubberband(
+            self._make_classic_style_pseudo_toolbar())
+
+
+class SaveFigureGTK(backend_tools.SaveFigureBase):
+    def trigger(self, *args, **kwargs):
+
+        class PseudoToolbar:
+            canvas = self.figure.canvas
+
+        return _NavigationToolbar2GTK.save_figure(PseudoToolbar())
```

Comment:
This doesn't work (whether on gtk3 or gtk4), as save_figure now resolves to the backend_bases' save_figure (which raises NotImplementedError).

Reply:
Oops, when I found the broken stuff, I tested on the original change, and forgot to do it after refactoring.